### PR TITLE
Add a top-level `yarn start` that watches all packages.

### PR DESCRIPTION
**Pull request by iameli**
**What does this pull request do? Explain your changes. (required)**
This PR adds `yarn start` commands to all of our packages that didn't already have them. It also adds a top-level `yarn start` command that fires off all of the sub-packages concurrently. This streamlines development a bit — you can have all of the packages compiling in one window.

It doesn't include a script for `merkle-miner` because it's mostly deprecated or one for `subgraph` because graph deploy takes a long time and you wouldn't want to fire it off until you were sure you were ready.

**Specific updates (required)**

- Added start scripts using `npm-run-all` and `nodemon` in one instance
- Added a top-level `yarn start` script that fires off all the other ones with Lerna
- Cleaned up some `package.json` logic using `npm-run-all`
- Adds a `depcheck` script that verifies unnecessary dependencies
- Removes some unnecessary internal monorepo dependencies
- Removes all unnecessary usage of `npx` in scripts

**How did you test each of these updates (required)**
Ran `yarn start` at the top level and ensured that all packages were compiling properly.

**Checklist:**

- [x] I have read the **CONTRIBUTING** document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,14 +41,20 @@ This project requires `node >=8.0.0` and `yarn >=1.0.0`. A unix shell is also re
...
-To get started, clone the repo and install its dependencies
+To get started, clone the repo and install its dependencies:
 
 ```bash
 git clone https://github.com/livepeer/livepeerjs.git
 cd livepeerjs
 yarn
 ```
 
+For development purposes there's a top-level `start` script that will watch and continiously compile all packages concurrently:
```
Comment (yondonfu) [inline README.md]: "continiously" -> "continuously"?

**Comment by iameli** [comment]
Updated to resolve merge conflicts with #333

**Review by Randy1Burrell — APPROVED** [review]
LGTM!

**Review by yondonfu — APPROVED** [review]
LGTM

**Pull request merged**